modules/nf-scil/betcrop/fslbetcrop/main.nf

```diff
@@ -23,31 +23,49 @@ process BETCROP_FSLBETCROP {
     def prefix = task.ext.prefix ?: "${meta.id}"
 
     def b0_thr = task.ext.b0_thr ? "--b0_thr " + task.ext.b0_thr : ""
-    def bet_dwi_f = task.ext.bet_dwi_f ? "-f " + task.ext.bet_dwi_f : ""
+    def bet_f = task.ext.bet_f ? "-f " + task.ext.bet_f : ""
+    def size_dil = task.ext.size_dil ? task.ext.size_dil : ""
 
     """
     export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
     export OMP_NUM_THREADS=1
     export OPENBLAS_NUM_THREADS=1
 
-    scil_extract_b0.py $dwi $bval $bvec ${prefix}__b0.nii.gz --mean \
-        $b0_thr --force_b0_threshold
-    bet ${prefix}__b0.nii.gz ${prefix}__b0_bet.nii.gz -m -R $bet_dwi_f
-    scil_image_math.py convert ${prefix}__b0_bet_mask.nii.gz ${prefix}__b0_bet_mask.nii.gz --data_type uint8 -f
-    mrcalc $dwi ${prefix}__b0_bet_mask.nii.gz -mult ${prefix}__dwi_bet.nii.gz -quiet -nthreads 1
+    if [[ -f "$bval" ]]
+    then
+        scil_extract_b0.py $image $bval $bvec ${prefix}__b0.nii.gz --mean \
+            $b0_thr --force_b0_threshold
 
-    scil_crop_volume.py $dwi ${prefix}__dwi_bet_cropped.nii.gz -f \
-        --output_bbox ${prefix}__dwi_boundingBox.pkl -f
-    scil_crop_volume.py ${prefix}__b0_bet_mask.nii.gz ${prefix}__dwi_bet_cropped_mask.nii.gz -f\
-        --input_bbox ${prefix}__dwi_boundingBox.pkl -f
-    scil_image_math.py convert ${prefix}__dwi_bet_cropped_mask.nii.gz ${prefix}__dwi_bet_cropped_mask.nii.gz \
-        --data_type uint8 -f
+        bet ${prefix}__b0.nii.gz ${prefix}__b0_bet.nii.gz -m -R $bet_f
+        scil_image_math.py convert ${prefix}__b0_bet_mask.nii.gz ${prefix}__image_bet_mask.nii.gz --data_type uint8 -f
+        mrcalc $image ${prefix}__b0_bet_mask.nii.gz -mult ${prefix}__image_bet.nii.gz -quiet -nthreads 1
+
+    else
+        bet $image ${prefix}__image_bet.nii.gz -m -R $bet_f
+        scil_image_math.py convert ${prefix}__image_bet_mask.nii.gz ${prefix}__image_bet_mask.nii.gz --data_type uint8 -f
+    fi
+
+    if [ "$task.ext.dilate" = "true" ];
+    then
+        scil_image_math.py dilation ${prefix}__image_bet_mask.nii.gz $size_dil ${prefix}__image_bet_mask.nii.gz --data_type uint8 -f
+    fi
+
+    if [ "$task.ext.crop" = "true" ];
+    then
+        scil_crop_volume.py ${prefix}__image_bet.nii.gz ${prefix}__image_bet.nii.gz -f \
+            --output_bbox ${prefix}__image_boundingBox.pkl -f
+        scil_crop_volume.py ${prefix}__image_bet_mask.nii.gz ${prefix}__image_bet_mask.nii.gz -f\
+            --input_bbox ${prefix}__image_boundingBox.pkl -f
```

Comment:
I think this should be the other way around, because if you dilate the mask beforehand but then use the bounding box from the non-dilated image, isn't there a chance the mask gets cropped? It might be safer to crop the mask first, output the bbox, then crop the image with this bbox.

Reply:
Indeed, good point.